### DISCUSSION TITLE
Stockfish on Raspberry pi 5

### DISCUSSION THREAD

**Discussion by paolo2504** [official-stockfish/Stockfish#5408] [Q&A]
At present it seems to me that latest Stockfish 16.1 is not available Raspberry pi 5 64bit. However i noticed that stockfish engine 10 aarch is available in the Engines folder in Arena 3.9 beta package.
Two questions:
 is r10 the latest aarch available for raspberry (arm76) ?
 no gui packaged 16.1 version for raspberry pi 5 as well ?
Thanks & regards.
paolo

**Comment by Disservin** [comment]
The armv8 versions should work for raspberry pi 5 https://github.com/official-stockfish/Stockfish/releases.

**Reply by paolo2504** [reply]
tried armv8 and armv7 in Scid. Error is "error writing file 8: broken pipe".

**Reply by vondele** [reply]
what happens is you just execute the binary:
`./stockfish go depth 10` 
(or the proper path to your downloaded binary?)

**Reply by paolo2504** [reply]
nothing happens : no errors and of course no gui.

i just tried to compile it and it works in scid and in Arena.
I downloaded Stockfish_16.1.zip, unzippped it and compiled with command make build ARCH=armv8. 
A 66.2 mb executable is generated and it works. Also make build ARCH=armv7 does work. 

On the contrary, the executable stockfish-android-armv8 doesn't work and generates errors when installed in Scid or Arena.
I tried other files (armv7... neon... avx2) and noe of them work on rpi 5. 
Definitely, compiling the source 16.1 as armv8 works fine in rpi 5 64bit.
regards.

**Comment by paolo2504** [comment]
Thanks Disservin. I tried android-armv8 already. 
launching the executable nothing happens. 
no error any case. Are this versions without gUI ?
I tried using scid and Arena pointing at the extracted executable but I am getting a errors. 
maybe I’m doing something wrong..

I see that the r10 engine of stock fish is very different in size so I suppose that using the executable of the 16.1 package (67mb) is a mistake.

**Reply by Disservin** [reply]
> Are this versions without gUI ?

We don't develop a GUI

> Stockfish does not include a graphical user interface (GUI) that is required to display a chessboard and to make it easy to input moves. These GUIs are developed independently from Stockfish and are available online. Read the documentation for your GUI of choice for information about how to use Stockfish with it.

https://github.com/official-stockfish/Stockfish

**Comment by paolo2504** [comment]
Luca-256, thank you for you kind reply. 
I confirm the release ...android-armv8 doesn't work on rpi5 either with SCID and Arena-linux.
Solution is to compile it fom the src folder with make build ARCH=armv8; the resulting executable is recognized by SCID and Arena.
regards.

**Reply by paolo2504** [reply]
Thanks Luca for your suggestion. I don’t know (for the moment…🙄) what are “dot products” but I’ll give a try for sure and make new compile as per your suggestion. 
regards.

**Reply by paolo2504** [reply]
hi Luca, they both work. See attached log of both makes.

[log make stockfish.txt](https://github.com/user-attachments/files/16443641/log.make.stockfish.txt)
